server/src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -47,6 +44,27 @@ public static boolean isEmpty(Object[] array) {
         return array == null || array.length == 0;
     }
 
+    public static <T> void unique(List<T> list, Comparator<T> cmp) {
```

Comment:
Add javadocs that the list must be sorted according to the given comparator?
Also a bit of a nit-pick, but since this modifies the list in-place, I feel like naming the method after a verb would be more appropriate?

Reply:
Changed to `uniquify`